docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -238,17 +238,174 @@ the strategy employed by the Example HTTP connector.
+      ``Int128`` for precision greater than 19
+  * - ``VARCHAR``
+    - ``Slice``
+  * - ``CHAR``
+    - ``Slice``
+  * - ``VARBINARY``
+    - ``Slice``
+  * - ``JSON``
+    - ``Slice``
+  * - ``DATE``
+    - ``long``
+  * - ``TIME(P)``
+    - ``long``
+  * - ``TIME WITH TIME ZONE``
+    - ``long`` for precision up to 9;
+      ``LongTimeWithTimeZone`` for precision greater than 9
+  * - ``TIMESTAMP(P)``
+    - ``long`` for precision up to 6;
+      ``LongTimestamp`` for precision greater than 6
+  * - ``TIMESTAMP(P) WITH TIME ZONE``
+    - ``long`` for precision up to 3;
+      ``LongTimestampWithTimeZone`` for precision greater than 3
+  * - ``INTERVAL YEAR TO MONTH``
+    - ``long``
+  * - ``INTERVAL DAY TO SECOND``
+    - ``long``
+  * - ``ARRAY``
+    - ``Block``
+  * - ``MAP``
+    - ``Block``
+  * - ``ROW``
+    - ``Block``
+  * - ``IPADDRESS``
+    - ``Slice``
+  * - ``UUID``
+    - ``Slice``
+  * - ``HyperLogLog``
+    - ``Object``
+  * - ``P4HyperLogLog``
+    - ``Object``
+  * - ``SetDigest``
+    - ``Object``
+  * - ``QDigest``
+    - ``Object``
+  * - ``TDigest``
+    - ``Object``
+
+The ``RecordCursor.getType(int field)`` method returns the SQL type for a field
+and the field value is returned by one of the following methods, matching
+the carrier type:
+
+* ``getBoolean(int field)``
+* ``getLong(int field)``
+* ``getDouble(int field)``
+* ``getSlice(int field)``
+* ``getObject(int field)``
+
+Values for the ``timestamp(p) with time zone`` and ``time(p) with time zone``
+types of regular precision can be converted into ``long`` using static methods
+from the ``io.trino.spi.type.DateTimeEncoding`` class, like ``pack()`` or
+``packDateTimeWithZone()``.
+
+UTF-8 encoded strings can be converted to Slices using
+the ``Slices.utf8Slice()`` static method.
 
-Given a split and a list of columns, the record set provider is
-responsible for delivering data to the Trino execution engine.
-It creates a ``RecordSet``, which in turn creates a ``RecordCursor``
-that's used by Trino to read the column values for each row.
+.. note::
+
+  The ``Slice`` class is provided by the ``io.airlift:slice`` package.
+
+``Int128`` objects can be created using the ``Int128.valueOf()`` method.
+
+The following example creates a block for an ``array(varchar)``  column:
+
+.. code-block:: java
+
+    private Block encodeArray(List<String> names)
+    {
+        BlockBuilder builder = VARCHAR.createBlockBuilder(null, names.size());
+        for (String name : names) {
+            if (name == null) {
+                builder.appendNull();
+            }
+            else {
+                VARCHAR.writeString(builder, name);
+            }
+        }
+        return builder.build();
+    }
+
+The following example creates a block for a ``map(varchar, varchar)`` column:
+
+.. code-block:: java
+
+    private Block encodeMap(Map<String, ?> map)
+    {
+        MapType mapType = new MapType(VARCHAR, VARCHAR, new TypeOperators());
```

Comment:
The type should be created via `TypeManager.getParameterizedType()`

Reply:
Done!